### PR TITLE
feat(onFoundTarget) hook executed if a new target is found

### PR DESCRIPTION
**Pull request by kybishop**

### CODE REVIEW

**Comment by kybishop** [comment]
Leaving as a WIP until I add some tests.

File: addon/components/ember-popper-base.js [hunk 1]
```diff
@@ -27,6 +27,12 @@ export default class EmberPopperBase extends Component {
   @property modifiers = null
 
   /**
+   * An optional function to be called when a new target is located.
+   * The target is passed in as an argument to the function.
+   */
+  @property onFoundTarget = function() {}
```
Comment (pzuraq) [inline addon/components/ember-popper-base.js]: This is probably not ideal, creates an extra closure. I would set it to `null` if it's expected to be a function.

Comment (kybishop) [reply addon/components/ember-popper-base.js]: fair enough. Was debating between what I have now and null, the disadvantage being checking for the existence of the action.

**Comment by kybishop** [comment]
@pzuraq re `sendAction`: I'm hesitant to switch to `sendAction` since the bubble effect likely has a small performance hit and we'll eventually move away from pre-1.13 support entirely.

Since `sendAction` bubbles up until it finds something while closure actions don't, switching to closure actions at a later date might bork users that were depending on `sendAction`'s wonky behavior.

Hope you don't mind me merging this for now since we can eventually backport <1.13 support if needed.

**Comment by pzuraq** [comment]
`sendAction` doesn't actually bubble in components, only in Routes and Controllers IIRC. For components, it just guards the action (sends it if it exists, does nothing if it doesn't).

**Comment by kybishop** [comment]
@pzuraq interesting. I'll merge a PR to switch to `sendAction`

**Pull request merged**